File: src/main/java/me/pljr/pljrapi/utils/MaterialUtil.java
package me.pljr.pljrapi.utils;

import com.cryptomorin.xseries.XMaterial;
import org.bukkit.Material;

public class MaterialUtil {
    /**
     * Checks if String is a {@link Material}.
     *
     * @param value String that should be checked for Material
     * @return True if value is Material, false if otherwise
     */
    public static boolean isMaterial(String value){
        for (Material material : Material.values()){
            if (material.toString().equalsIgnoreCase(value)){
                return true;
            }
        }
        return false;
    }

    /**
     * Checks if String is a {@link XMaterial}.
     *
     * @param value String that should be checked for XMaterial
     * @return True if value is XMaterial, false if otherwise
     */
    public static boolean isXMaterial(String value){
        return XMaterial.matchXMaterial(value).isPresent();
    }
}

File: src/main/kotlin/me/pljr/pljrapispigot/util/MaterialUtil.kt
package me.pljr.pljrapispigot.util

import org.bukkit.Material

/**
 * Checks if String is a [Material].
 *
 * @param value String that should be checked for Material
 * @return True if value is Material, false if otherwise
 */
fun isMaterial(value: String): Boolean {
    for (material in Material.values()) {
        if (material.toString().equals(value, true)) {
            return true
        }
    }
    return false
}